thoth/storages/graph/postgres.py

```diff
@@ -1186,21 +1186,61 @@ def get_python_package_index_urls(self, enabled: bool = None) -> set:
 
         return set(item[0] for item in query.with_entities(PythonPackageIndex.url).distinct().all())
 
-    def get_python_packages_for_index(self, index_url: str) -> Set[str]:
+    def get_python_packages_per_index(self, index_url: str, distinct: bool = False) -> Dict[str, List[str]]:
         """Retrieve listing of Python packages known to graph database instance for the given index."""
-        return set(
-            item[0]
-            for item in self._session.query(PythonPackageVersion)
+        query = (
+            self._session.query(PythonPackageVersion)
             .join(PythonPackageIndex)
             .filter(PythonPackageIndex.url == index_url)
             .with_entities(PythonPackageVersion.package_name)
-            .distinct()
-            .all()
         )
 
-    def get_python_packages(self) -> Set[str]:
-        """Retrieve listing of all Python packages known to graph database instance."""
-        return set(item[0] for item in self._session.query(PythonPackageVersionEntity.package_name).distinct().all())
+        if distinct:
+            query.distinct()
+
+        query = query.all()
+
+        return {index_url: [item[0] for item in query]}
+
+    def get_python_package_version_entities_count_all(
+        self,
+        *,
+        start_offset: int = 0,
+        count: int = 100,
+        python_version: str = None,
+        distinct: bool = False,
+    ) -> int:
+        """Retrieve listing of all Python package versions entity with known index to graph database instance."""
+        query = (
```

Comment:
This looks like syntax error!? No matching `)` for this bracket - here and bellow.